test/e2e/io_soak/io_soak_test.go

```diff
@@ -44,29 +26,79 @@ func TestIOSoak(t *testing.T) {
 	common.InitTesting(t, "IO soak test, NVMe-oF TCP and iSCSI", "io-soak")
 }
 
-func monitor(errC chan<- error) {
-	logf.Log.Info("IOSoakTest monitor, checking mayastor and test pods")
-	for {
-		time.Sleep(30 * time.Second)
-		err := common.CheckPods(common.NSMayastor)
+func monitor() error {
+	var err error
+	var failedJobs []string
+	activeJobMap := make(map[string]IoSoakJob)
+	for _, job := range jobs {
+		activeJobMap[job.getPodName()] = job
+	}
+
+	podsRunning := 0
+	podsSucceeded := 0
+	podsFailed := 0
+
+	logf.Log.Info("IOSoakTest monitor, checking mayastor and test pods", "jobCount", len(activeJobMap))
+	for ; len(activeJobMap) !=0 && len(failedJobs) == 0; {
+		time.Sleep(29 * time.Second)
+		err = common.CheckPods(common.NSMayastor)
```

Comment:
I might rename CheckPods to CheckPodRestarts

Reply:
I've renamed to CheckTestPodHealth since the function also checks pod status phase.